File: srcs/tools.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   tools.c                                            :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: becaraya <becaraya@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2019/12/07 14:22:25 by becaraya          #+#    #+#             */
/*   Updated: 2020/01/28 11:57:57 by ydemange         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include "doom_nukem.h"

t_point			itop(int x, int y)
{
	t_point		res;

	res.x = x;
	res.y = y;
	return (res);
}

int				inr(t_point src_a, t_point src_b, t_point cmp)
{
	if (src_a.x < cmp.x && src_a.y < cmp.y
		&& src_b.x > cmp.x && src_b.y > cmp.y)
		return (1);
	return (0);
}

unsigned int	nb_wall(t_al *al)
{
	t_sector	*tmp_s;
	int			i;

	tmp_s = al->sect;
	i = 0;
	while ((al->nb_sec - i) > al->edit.index_sect)
	{
		tmp_s = tmp_s->next;
		i++;
	}
	return ((tmp_s) ? tmp_s->nb_wal : 0);
}

int				cocmp(SDL_Rect *p1, SDL_Rect *p2)
{
	if (p1->x == p2->x && p1->y == p2->y)
		return (1);
	return (0);
}

File: web version/src/tools.js
function rotate(al) {
    if (al.k.left) {
        al.play.dir = subAngle(al.play.dir, (LOOK_SENS * D_2PI / 2500) * al.dtime / 1000000);
    }
    if (al.k.righ) {
        al.play.dir = addAngle(al.play.dir, (LOOK_SENS * D_2PI / 2500) * al.dtime / 1000000);
    }
    if (al.k.up) {
        al.play.horizon -= LOOK_SENS * al.dtime / 1000000;
    }
    if (al.k.down) {
        al.play.horizon += LOOK_SENS * al.dtime / 1000000;
    }
    if (al.play.horizon < -HORIZON_LIMIT) {
        al.play.horizon = -HORIZON_LIMIT;
    }
    if (al.play.horizon > HORIZON_LIMIT) {
        al.play.horizon = HORIZON_LIMIT;
    }
}

function itop(x, y) {
    return { x, y };
}

function inr(srcA, srcB, cmp) {
    return srcA.x < cmp.x && srcA.y < cmp.y && srcB.x > cmp.x && srcB.y > cmp.y ? 1 : 0;
}

function nbWall(al) {
    let tmpS = al.sect;
    let i = 0;

    while (al.nbSec - i > al.edit.indexSect) {
        tmpS = tmpS.next;
        i++;
    }

    return tmpS ? tmpS.nbWal : 0;
}

function coCmp(p1, p2) {
    return p1.x === p2.x && p1.y === p2.y ? 1 : 0;
}

function subAngle(angle1, angle2) {
    return (angle1 - angle2 + D_2PI) % D_2PI;
}

function addAngle(angle1, angle2) {
    return (angle1 + angle2) % D_2PI;
}